File: framework/test2/validate/numeric/numeric.py
#!/usr/bin/env python3
from collections import Counter, defaultdict, deque
import heapq
import json
import logging
import math
import random
import time
from typing import *
import sys

from soda.leetcode.bitree import *
from soda.leetcode.graph import *
from soda.leetcode.linklist import *
from soda.leetcode.nest import *
from soda.unittest.util import init_logging

logger = logging.getLogger(__name__)

# step [1]: implement class Solution
class Solution:
    def multiply(self, a: List[float], b: List[float]) -> List[float]:
        return list(i * j for i, j in zip(a, b))

if __name__ == '__main__':
    init_logging()
    from soda.unittest import Validators
    from soda.unittest.work import TestWork

    # step [2]: setup function
    # Attention! FUNCTION must use type hint, including arguments and return type
    work = TestWork(Solution().multiply)
    # OR use struct tester
    # work = TestWork.forStruct(CLASS)

    # step [3]: setup other options
    work.validator = Validators.forList(float, True)
    work.compareSerial = True
    work.run()

File: framework/test2/validate/numeric/Numeric.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.work.TestWork;

import static soda.unittest.LoggerHelper.logger;

class Solution {
    public double[] multiply(double[] a, double[] b) {
        var res = new double[a.length];
        for (int i = 0; i < res.length; ++i) {
            res[i] = a[i] * b[i];
        }
        return res;
    }
}

public class Numeric implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "multiply");
        // var work = TestWork.forStruct(Struct.class);
        work.setValidator(Validators.forDoubleArray(true));
        work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Numeric().get().run();
    }

}
